interfaces/cython/cantera/test/test_thermo.py

```diff
@@ -1676,15 +1676,74 @@ def test_slicing_ndim(self):
         self.assertArrayNear(col3.T, 900*np.ones(2))
         self.assertArrayNear(row2.T, 900*np.ones(5))
 
-    def test_extra(self):
+    def test_extra_create_by_dict(self):
         extra = OrderedDict([('grid', np.arange(10)),
                              ('velocity', np.random.rand(10))])
         states = ct.SolutionArray(self.gas, 10, extra=extra)
         keys = list(states._extra.keys())
         self.assertEqual(keys[0], 'grid')
-
-        with self.assertRaises(ValueError):
-            states = ct.SolutionArray(self.gas, extra=['creation_rates'])
+        self.assertArrayNear(states.grid, np.arange(10))
+
+    def test_extra_no_shape(self):
+        # The shape of the value for "prop" here is (), which is falsey
+        # and causes the use of np.full()
+        states = ct.SolutionArray(self.gas, 3, extra={"prop": 1})
+        self.assertEqual(states.prop.shape, (3,))
+        self.assertArrayNear(states.prop, np.array((1, 1, 1)))
+
+        # Check a multidimensional SolutionArray
+        states = ct.SolutionArray(self.gas, (2, 2), extra={"prop": 3})
+        self.assertEqual(states.prop.shape, (2, 2))
+        self.assertArrayNear(states.prop, np.array(((3, 3), (3, 3))))
+
+    def test_extra_wrong_shape(self):
+        with self.assertRaisesRegex(ValueError, "Unable to map"):
+            ct.SolutionArray(self.gas, (3, 3), extra={"prop": np.arange(3)})
+
+    def test_extra_create_by_iterable(self):
+        states = ct.SolutionArray(self.gas, extra=("prop1"))
+        self.assertEqual(states.prop1.shape, (0,))
+
+        # An integer is not an iterable, and only bare strings are
+        # turned into iterables
+        with self.assertRaisesRegex(ValueError, "Extra properties"):
+            ct.SolutionArray(self.gas, extra=2)
+
+    def test_extra_not_string(self):
+        with self.assertRaisesRegex(TypeError, "is not a string"):
+            ct.SolutionArray(self.gas, extra=[1])
+
+    def test_extra_reserved_names(self):
+        with self.assertRaisesRegex(ValueError, "name is already used"):
+            ct.SolutionArray(self.gas, extra=["creation_rates"])
+
+        with self.assertRaisesRegex(ValueError, "name is already used"):
+            ct.SolutionArray(self.gas, extra={"creation_rates": 0})
+
+    def test_extra_create_by_string(self):
```

Comment:
Same here. I believe allowing for this may create a slew of problems elsewhere.

Reply:
This creates one empty `extra` item by passing a single string, the values in the array are not strings. I believe this should be left.

Reply:
Correct.